[bug]: Remote Signer lnd shutdown panic

### Background

i deployed lnd as remote signer. when it shutdown it will panic.

### Your environment

* 0.19.0-beta
* Macos
* nochainbackend

### Steps to reproduce
1. Set lnd  as a remote signer, nochainbackend in conf
2. unlock wallet
3. let it run a while then shutdown


### Error
```
2025-06-18 16:37:27.939 [INF] CHIO: BlockbeatDispatcher is starting with 1 consumer queues
2025-06-18 16:37:31.135 [INF] LTND: Received interrupt
2025-06-18 16:37:31.135 [INF] LTND: Shutting down...
2025-06-18 16:37:31.135 [INF] LTND: Gracefully shutting down.
2025-06-18 16:37:31.135 [INF] CHIO: BlockbeatDispatcher is stopping
2025-06-18 16:37:31.135 [INF] NANN: Channel Status Manager shutting down...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10152dc80]

goroutine 417 [running]:
github.com/lightningnetwork/lnd/chainio.(*BlockbeatDispatcher).log(...)
	/Users/user/code/lnd/chainio/dispatcher.go:138
github.com/lightningnetwork/lnd/chainio.(*BlockbeatDispatcher).dispatchBlocks(0x140009ba5a0, 0x140008b0f60)
	/Users/user/code/lnd/chainio/dispatcher.go:198 +0x420
created by github.com/lightningnetwork/lnd/chainio.(*BlockbeatDispatcher).Start in goroutine 238
	/Users/user/code/lnd/chainio/dispatcher.go:122 +0x1e0
```

--- a/chainio/dispatcher.go
+++ b/chainio/dispatcher.go
@@ -135,6 +135,14 @@ func (b *BlockbeatDispatcher) Stop() {
 }
 
 func (b *BlockbeatDispatcher) log() btclog.Logger {
+	// There's no guarantee that the `b.beat` is initialized when the
+	// dispatcher shuts down, especially in the case where the node is
+	// running as a remote signer, which doesn't have a chainbackend. In
+	// that case we will use the package logger.
+	if b.beat == nil {
+		return clog
+	}
+
 	return b.beat.logger()
 }
 

--- a/docs/release-notes/release-notes-0.19.2.md
+++ b/docs/release-notes/release-notes-0.19.2.md
@@ -26,6 +26,9 @@
 - [Fixed](https://github.com/lightningnetwork/lnd/pull/9921) a case where the
   spending notification of an output may be missed if wrong height hint is used.
 
+- [Fixed](https://github.com/lightningnetwork/lnd/pull/9962) a case where the
+  node may panic if it's running in the remote signer mode.
+
 # New Features
 
 ## Functional Enhancements

